### ISSUE TITLE
TypeError: Illegal buffer - on requests

### ISSUE THREAD

**Issue by rohanp** [stoffern/pokemongo-api#84]
Getting this error every time I attempt to login 

```
TypeError: Illegal buffer
    at TypeError (native)
    at Function.module.exports.ByteBuffer.wrap (/Users/rohanp/hack/pokemongo-api/node_modules/bytebuffer/dist/bytebuffer-node.js:346:19)
    at Function.Message.decode (/Users/rohanp/hack/pokemongo-api/node_modules/protobufjs/dist/protobuf.js:2930:84)
    at src/API.js:34:55
    at Array.map (native)
    at Connection._callee$ (src/API.js:28:14)
    at tryCatch (/Users/rohanp/hack/pokemongo-api/node_modules/regenerator-runtime/runtime.js:62:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/rohanp/hack/pokemongo-api/node_modules/regenerator-runtime/runtime.js:336:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Users/rohanp/hack/pokemongo-api/node_modules/regenerator-runtime/runtime.js:95:21)
    at step (src/API.js:8:1)
```

**Comment by gallexme** [comment]
Show us your code

**Comment by rohanp** [comment]
I'm just using the example code. I'm having the same problem as #90. Do I need to generate new settings.js data? I know I'm not banned, because I can still log in on my phone

**Comment by stoffern** [comment]
@rohanp Google or PTR accounting?

**Comment by rohanp** [comment]
PTR. I was using multiple accounts, and all of them give the same error.

**Comment by stoffern** [comment]
What happends if you use one account only?

**Comment by rohanp** [comment]
I tried using one account at a time, but that is not working. I will try
creating a new account and using that.

**Comment by stoffern** [comment]
:+1:  do that and tell me how it works..

**Comment by rohanp** [comment]
This it strange... The new account seems to work for some amount of time (immediately to a few minutes) then throws the same error. The old accounts throw the errors immediately every time.

**Comment by cscsomers** [comment]
I'm also experiencing this problem with both google and PTC accounts.

**Comment by stoffern** [comment]
please give me some more info on this if you can.
@rohanp if you can and want, email me an account login that has these problems i can do some debugging on it.
my email is kristoffer.klevenberg@gmail.com (i will tell you when i start and stop debugging on the account)

**Comment by rohanp** [comment]
done

**Comment by stoffern** [comment]
Ref: https://github.com/AHAAAAAAA/PokemonGo-Map/issues/2632

> Yesterday Niantic implemented a throttling system of 3 requests/second (I believe this is the limit, correct me if wrong) per account.
> 
> This throttle is why you are seeing less pokemon and receiving the error in console.


We can give this a quick fix in the api to allow requests every 3 seconds in the connection class.
It is possible to send multiple requests in one query to the api, so i think we wil setup a transport service that takes the full queue and queryes the db every 3 seconds
Give it 24h and i have fixed this! :smile: 

Meanwhile setting a wait for 3 secs will fix it:
```js

let objects = await Poke.GetMapObjects()
await new Promise(resolve => setTimeout(resolve, 3000))
let objects2 = await Poke.GetMapObjects()
```

cc @jimmywarting @gallexme

**Comment by stoffern** [comment]
This is now fixed in new commit: e56f27d3db6eb2d7c1be485978f3fe8897b10087
Added timer check so if you call function within 3 seconds it will automaticly wait for you and return MapObjects as fast as it can.
